WordPress/src/main/java/org/wordpress/android/designsystem/DesignSystemTheme.kt

```diff
@@ -0,0 +1,137 @@
+package org.wordpress.android.designsystem
+
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.material3.ColorScheme
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ProvideTextStyle
+import androidx.compose.material3.Surface
+import androidx.compose.material3.darkColorScheme
+import androidx.compose.material3.lightColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+
+private val localColors = staticCompositionLocalOf { extraPaletteLight }
+
+@Composable
+fun DesignSystemTheme(
+    isDarkTheme: Boolean = isSystemInDarkTheme(),
+    content: @Composable () -> Unit
+) {
+    DesignSystemThemeWithoutBackground(isDarkTheme) {
+        ContentInSurface(content)
+    }
+}
+
+@Composable
+fun DesignSystemThemeWithoutBackground(
+    isDarkTheme: Boolean = isSystemInDarkTheme(),
+    content: @Composable () -> Unit
+) {
+    val extraColors = if (isDarkTheme) {
+        extraPaletteDark
+    } else {
+        extraPaletteLight
+    }
+    CompositionLocalProvider (localColors provides extraColors) {
+        MaterialTheme(
+            colorScheme = if (isDarkTheme) {
+                darkColorScheme(
```

Comment:
You could probably lift this out like so...

val dynamicColorScheme = if (isSystemInDarkTheme) DarkColorScheme else LightColorScheme

    MaterialTheme(
        colorScheme = dynamicColorScheme,
        typography = Typography,
        shapes = Shapes,
        content = content
    )